### ISSUE TITLE
Proposal to add EIP-5022 to Shanghai

### ISSUE THREAD

**Issue by greenlucid** [ethereum/pm#519]
EIP-5022 link https://github.com/ethereum/EIPs/pull/5022

### tl;dr
Increase SSTORE when zero -> non-zero to `40_000` gas

### why
SSTORE is not priced correctly, its cost is way higher than its pricing.
Also, with https://github.com/ethereum/EIPs/pull/4844, rollup data won't even be competing with SSTOREs because they will be priced differently, which will not allow to slow down state growth.

### why not do it better (state expiry, 1559-like SSTORE pricing, state rent...)
Because that wouldn't be fast enough to be included in a fork as soon as possible. The cost of SSTOREs for the network is too different from its pricing 

### can this be done fast?
Yes. Opcode gas repricing has been done in the past, and even though the implementation is always manual, it is always streamlined. This is why this is a good contender for Shanghai inclusion, or even before.

### could this be done before Shanghai?
I would advocate to have this added on the merge or the (likely) difficulty bomb push before the merge, since a short term solution for state growth is needed. Shanghai appears to be a big fork. Waiting to get this included in Shanghai will mean another year or so of the network subsidizing SSTOREs.

**Comment by timbeiko** [comment]
@greenlucid note that on #616 we agreed to use Ethereum Magicians to discuss potential Shanghai EIPs async over the next few weeks. See [here](https://ethereum-magicians.org/t/shanghai-core-eip-consideration/10777/) for more info.

**Comment by timbeiko** [comment]
We finalized the scope for Shanghai on ACD151, and discussions about Cancun inclusion should be moved to https://ethereum-magicians.org/t/cancun-eip-consideration/12060
